### PR TITLE
Support comma as a hotkey

### PR DESCRIPTION
**Pull request by colinwm**
Right now, we use comma as a separator between hotkeys, but this precludes us from using commas a hotkey!

My solution is to allow specifying a comma hotkey via `data-hotkey="comma"`, which gets rewritten as `,`. I'm open to alternative strategies though, let me know what you think.

### CODE REVIEW

**Comment by keithamus** [comment]
Thanks for the PR @colinwm! We're trying to keep consistency with `event.key` to make the translation straightforward. What do you think about improving the parser instead, to catch cases like this? Perhaps it's possible to make this intuitive enough while also making `,` as both the separator and the key symbol?

File: test/test.js [hunk 1]
```diff
@@ -113,6 +113,12 @@ describe('hotkey', function () {
       document.dispatchEvent(new KeyboardEvent('keydown', {shiftKey: true, code: 'KeyB', key: 'B'}))
       assert.notInclude(elementsActivated, 'button1')
     })
+
+    it('supports comma as a hotkey', function () {
+      setHTML('<button id="button1" data-hotkey="comma">Button 1</button>')
```
Comment (theinterned) [inline test/test.js:118]: This is great to see! Can you please also add a test for the comma in concert with some modifier keys such as `Control`?

Comment (colinwm) [reply test/test.js:118]: Done

**Comment by colinwm** [comment]
> Perhaps it's possible to make this intuitive enough while also making , as both the separator and the key symbol?

@keithamus I think it'd be tough to make a parser that would correctly parse commas as both separators and key symbols. E.g. the parser would need to be able to handle some pretty weird cases, like:

`,` --> comma
`,,.` --> comma or period
`Ctrl+Meta+,,x` --> ctrl+meta+comma or x

which IMO is kind of unreadable and would require a parser that might contain some bugs. But let me know if you really want me to implement it.

**Review by theinterned — COMMENTED** [review]
Thank you for adding these extra tests. Great!

I will let @keithamus way in on the `,` vs `coma` thing before  considering approving. One thing I will say is that if we go with the string, it would be good to Titlecase it to be consistent with the rest of the syntax. `Comma` is also the name of the corresponding `event.code` https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code/code_values

**Comment by colinwm** [comment]
@theinterned updated to use title case. @keithamus let me know what you think!

**Comment by theinterned** [comment]
@colinwm You can wait to hear what @keithamus has to say, but if the `Comma` route works, then please also update the `README` to mention it. There's a **"Hotkey string format"** section that it would make sense in here https://github.com/github/hotkey#hotkey-string-format

**Comment by keithamus** [comment]
> `,` --> comma
> `,,.` --> comma or period
> `Ctrl+Meta+,,x` --> ctrl+meta+comma or x

None of these seem like ambiguous cases to me. They're not especially nice to look at for sure, but they're not ambiguous. 

It'd be worth discovering if there ambiguous cases but I cannot think of one.

**Comment by colinwm** [comment]
@keithamus I've implemented a parser that can understand commas, PTAL. It's somewhat ugly but seems to work. Another annoying thing to consider is sequences, where commas can be used as keystrokes or separators. Although it's definitely very confusing, I believe it's not ambiguous. Examples:

`, a b,c` --> [Sequence(`,ab`), Hotkey(`c`)]
`a , b,c` --> [Sequence(`a,b`), Hotkey(`c`)]

Let me know what you think.

**Review by keithamus — APPROVED** [review]
I like this change. I like the parser. I agree the contrived examples are horrendous to look at but I think that this solves problems bigger than those!

I think what would take this PR from great to excellent is a few more test cases with the examples you've provided in the comments above. Plus a few more:

- `, , , ,`
- `Ctrl+x, , , `
- `,`

It also looks like there's some weird behaviour with ` ` now:

 - `expandHotKeyToEdges('    ')` (4 spaces) => `[ "", "", "", "" ]` (4 empty strings)

Not to lose sight of the original PR but perhaps this is something we could either handle (perhaps the output should be `[]`?) or throw?

**Comment by colinwm** [comment]
Thanks for the review @keithamus. I added a few more tests as you suggested and filtered out empty sequences/hotkeys. Will merge now
